core/src/main/java/com/linecorp/armeria/server/annotation/decorator/CorsDecorator.java

```diff
@@ -40,7 +40,12 @@
      * Allowed origins.
      * Sets this property to be {@code "*"} to allow any origin.
      */
-    String[] origins();
+    String[] origins() default {};
+
+    /**
+     * Specify allowed origins by regular expression.
+     */
+    String originRegex() default "";
```

Comment:
What do you think of taking a list of regex in case it is difficult to write all patterns in one regular expression?

Reply:
That's a good suggestion. 👍 Would you mind adding a patch for that? 😆

Reply:
Should we also allow list of regex for `CorsPolicy`?

Reply:
I think it's okay for now. We can add it later if we really need it.